include/lsst/afw/typehandling/GenericMap.h

```diff
@@ -409,8 +412,9 @@ class GenericMap {
      * type (which may be `void`). Through any combination of overloading or
      * templates, the visitor must accept values of the following types:
      *      * either `bool` or `bool const&`
-     *      * either `std::int32_t` or `std::int32_t const&`
-     *      * either `std::int64_t` or `std::int64_t const&`
+     *      * either `int` or `int const&`
+     *      * either `long` or `long const&`
+     *      * either `long long` or `long long const&`
```

Comment:
It's a pity this problem with type equivalency bleeds into the public interface; I'd have preferred to keep the interface in terms of `std::int32_t` and `std::int64_t`.  But I also don't see a good way around it; it's probably not worth fighting the language's native treatment of integer types to that extent, especially in the short term with DM-21327 on the horizon.  But if you do see a good way to do that, it might be worth ticketing.

Reply:
If I did, I would have done it. Honestly, even the existing level of overloading (e.g., `float` vs. `double`) is problematic.

Note that outside of a testing context, you could write these visitors with templates, and the distinction gets swept under the rug.

Reply:
Hmm... you know, I think we might be able to decouple the API from the `boost::variant` by providing an internal adapter that has explicit methods for each `variant` type but then delegates to the user-provided callable in a way that allows implicit conversions (I'm not completely sure this is possible, as the adapter would have to be templated on the callable, though not on the value type). That might be a worthwhile ticket.

Reply:
I'm pretty sure it's doable in C++; making it work with Python callable would at least be harder.  Worth a ticket, though I don't think I'd make it a priority right now.

Reply:
Written up as [DM-21382](https://jira.lsstcorp.org/browse/DM-21382).